content/en/docs/gke/deploy/upgrade.md

```diff
@@ -24,71 +38,167 @@ However, specific upgrades might need manual actions below.
+- Always recommend backing up your data before an upgrade.
+- Upgrading Kubeflow cluster can be a disruptive process, please schedule some downtime and communicate with your users.
+
+For upgrading from specific versions, there can be manual actions needed, so recommend referring to the specific sections below.
+
+Here are general instructions to upgrade Kubeflow:
+
+1.  The instructions below assume
+
+    - your current working directory is
+
+      ```bash
+      cd ${KF_DIR}
+      ```
+
+    - your kubectl uses a context that connects to your Kubeflow cluster
+      ```bash
+      # List your existing contexts
+      kubectl config get-contexts
+      # Use the context that connects to your Kubeflow cluster
+      kubectl config use-context ${KF_NAME}
+      ```
+
+1.  Edit the Makefile at `./Makefile` and change `MANIFESTS_URL` to point at the version of Kubeflow manifests you want to use
+
+    - Refer to the [kpt docs](https://googlecontainertools.github.io/kpt/reference/pkg/) for more info about supported dependencies
+
+1.  Update the local copies:
+
     ```bash
-    cd ${KF_DIR}/management
+    make update
     ```
 
-1.  Use your management cluster's kubectl context:
+1.  Redeploy:
 
     ```bash
-    # Look at all your contexts
-    kubectl config get-contexts
-    # Select your management cluster's context
-    kubectl config use-context ${MGMTCTXT}
-    # Verify the context connects to the cluster properly
-    kubectl get namespace
+    make apply
     ```
 
-    If you are using a different enviroment, you can always
-    reconfigure the context by:
+    To evaluate the changes before deploying them you can run `make hydrate` and then compare the contents
+    of `.build` to what is currently deployed.
 
-    ```bash
-    make create-context
-    ```
+### Upgrade Kubeflow cluster from v1.1 to v1.2
 
-1.  Check your existing config connector version:
-    ```bash
-    # For Kubeflow v1.1, it should be 1.15.1
-    $ kubectl get namespace cnrm-system -ojsonpath='{.metadata.annotations.cnrm\.cloud\.google\.com\/version}'
-    1.15.1
-    ```
-1.  Uninstall the old config connector in the management cluster:
-    ```bash
-    kubectl delete sts,deploy,po,svc,roles,clusterroles,clusterrolebindings --all-namespaces -l cnrm.cloud.google.com/system=true --wait=true
-    kubectl delete validatingwebhookconfiguration abandon-on-uninstall.cnrm.cloud.google.com --ignore-not-found --wait=true
-    kubectl delete validatingwebhookconfiguration validating-webhook.cnrm.cloud.google.com --ignore-not-found --wait=true
-    kubectl delete mutatingwebhookconfiguration mutating-webhook.cnrm.cloud.google.com --ignore-not-found --wait=true
-    ```
-    These commands uninstall the config connector without removing your resources.
-1.  Replace your `./Makefile` with the version in Kubeflow `v1.2.0`: https://github.com/kubeflow/gcp-blueprints/blob/v1.2.0/management/Makefile.
+1.  The instructions below assume
```

Comment:
```suggestion
1.  The instructions assume that:
```

Reply:
I think we need to keep this `below`, because in some docs, this sentence appears multiple times.